src/main/kotlin/com/atlassian/performance/tools/workspace/api/git/GitRepo.kt

```diff
@@ -29,14 +31,28 @@ interface GitRepo {
             if (descendant.isDirectory) {
                 descendant
                     .listFiles()
-                    .singleOrNull { it.name == ".git" }
-                    ?.let { return FileRepository(it) }
+                    ?.singleOrNull { it.name == ".git" }
+                    ?.let { return FileRepository(toGitFolder(it)) }
             }
-            val parent = descendant.parentFile
-            return when (parent) {
+            return when (val parent = descendant.parentFile) {
                 null -> null
                 else -> findInAncestors(parent)
             }
         }
+
+        private fun toGitFolder(file: File): File? {
+
+            if (file.isDirectory) {
+                return file
+            }
+
+            val line = file.bufferedReader().use { it.readLine() }
+            val pattern = "gitdir: (.*)".toRegex()
```

Comment:
`.git` can be a file 😮 
Today I learned 🧐

Reply:
So this is a bug fix. It's interesting for the consumers of the lib, so let's put it as `### Fixed` in the `CHANGELOG.md`

- [ ] changelog update